### PR TITLE
Bug 1867980: do not fail kubevirt-vmware deployment when CR contains a space in URL

### PR DESCRIPTION
**Pull request by suomiy**
@pkliczewski @mareklibra @nyoxi @nertpinx  please review

### CODE REVIEW

File: kubevirt-vmware/pkg/controller/v2vvmware/client.go [hunk 1]
```diff
@@ -139,6 +140,10 @@ func NewClient(ctx context.Context, credentials *LoginCredentials) (*Client, err
 
 	log.Info(fmt.Sprintf("NewClient, user: '%s', host: '%s'", credentials.username, credentials.host))
 
+	if strings.IndexFunc(credentials.host, unicode.IsSpace) >= 0 {
```
Comment (mareklibra) [inline kubevirt-vmware/pkg/controller/v2vvmware/client.go:143]: What about trimming it first? That should not cause any semantics issues.

Comment (suomiy) [reply kubevirt-vmware/pkg/controller/v2vvmware/client.go:143]: it depends on what we want to do:
1. fail fast and let user know that s/he has a faulty secret (which might be used by other places) 
2. be lenient and trim it first

thoughts?

Comment (pkliczewski) [reply kubevirt-vmware/pkg/controller/v2vvmware/client.go:143]: I think we should trim

Comment (suomiy) [reply kubevirt-vmware/pkg/controller/v2vvmware/client.go:143]: ok, done

**Pull request merged**